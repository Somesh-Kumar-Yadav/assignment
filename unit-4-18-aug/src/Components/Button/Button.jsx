import React from "react"
import styles from './Button.module.css'

export default function Button({ label, onClick }) {
    return (
        <div data-testid="btn" onClick={onClick} className={styles.container}>
                {label}
        </div>
    )
}