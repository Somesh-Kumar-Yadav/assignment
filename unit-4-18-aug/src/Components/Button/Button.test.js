import { render, screen, fireEvent } from "@testing-library/react";
import Button from "./Button";
import renderer from "react-test-renderer";
const testIds = {
	button: "btn",
};

describe("Button testing", () => {
	test("button should render", () => {
		const text = "Click Me";
		render(<Button label={text} />);
		const btn = screen.getByText(text);
		expect(btn).toBeInTheDocument();
	});
	test("button should render with correct label", () => {
		const text = "Click Me";
		render(<Button label={text} />);
		const btn = screen.getByTestId("btn");
		expect(btn.textContent).toBe(text);
	});
	it("Should invoke a function when you click on it", () => {
		const fn = jest.fn();
		const text = "Click Me";
		render(<Button label={text} onClick={fn} />);
		const btn = screen.getByTestId(testIds.button);
		fireEvent.click(btn);

		expect(fn).toHaveBeenCalled();
		expect(fn).toHaveBeenCalledTimes(1);
		fireEvent.click(btn);
		fireEvent.click(btn);
		fireEvent.click(btn);
		expect(fn).toHaveBeenCalledTimes(4);
	});
	it("Should matchthe snapshot", () => {
		const tree = renderer.create(
			<Button data-testid="btn" onClick={() => {}} label="Button" />
		);
		expect(tree).toMatchSnapshot();
	});
});
