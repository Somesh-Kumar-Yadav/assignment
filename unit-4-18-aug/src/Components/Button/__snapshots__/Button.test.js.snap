// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button testing Should matchthe snapshot 1`] = `
<div
  className="container"
  data-testid="btn"
  onClick={[Function]}
>
  Button
</div>
`;
